src/service/print/csv.js

```diff
@@ -0,0 +1,298 @@
+const { getData } = require('./csvImporter')
+const JSON2CSV = require('json2csv').parse
+var cheerio = require('cheerio')
+​
+const buildCSVWithCallback = async (id, callback) => {
+  let error = false
+  let errorMsg = ''
+  let totalMarks = 0
+  getData(id)
+    .then(async data => {
+      if (data.error) {
+        callback(null, data.error, data.errorMsg,null)
+      } else {
+        const subject = data.paperData.subject[0]
+        const grade = data.paperData.gradeLevel[0]
+        const examName = data.paperData.name
+        // console.log("paperdata:",data.paperData);
+  
+        data.sectionData.forEach(d => {
+          d.questions.forEach((element, index) => {
+            const marks = parseInt(d.section.children[index].marks)
+            if (!isNaN(marks)) totalMarks += marks
+          })
+        })
+​
+        const questionPaperContent = []
+        let questionCounter = 0
+        for (const d of data.sectionData) {
+          for (const [index, question] of d.questions.entries()) {
+            questionCounter += 1
+            let questionContent
+            let blooms  
+            let learningOutcome
+           
+            if (question.category === 'MCQ') {
+               if(question.learningOutcome[0] === undefined) {
+                 learningOutcome = ""
+               } else {
+                 learningOutcome = question.learningOutcome[0]
+               }
+               if(question.bloomsLevel[0] === undefined) {
+                blooms = ""
+              } else {
+                blooms = question.bloomsLevel[0]
+              }
+              
+          if(blooms === "Remember"){
+            blooms = "Knowledge"
+          }  
+          else if(blooms === "Understand"){
+            blooms = "Understanding"
+          } 
+          else if(blooms === "Apply") {
+           
+            blooms = "Application"
+          }
+          console.log("Changed value:",blooms);
+             
+​
+              let chaperName = question.topic[0]
```

Comment:
Similar null check can be added here  for `topic` as needed for `learningOutcome` and `bloomsLevel`